addons/fleet/models/fleet_vehicle.py

```diff
@@ -25,7 +25,7 @@ def _get_default_state(self):
         help='License plate number of the vehicle (i = plate number for a car)')
     vin_sn = fields.Char('Chassis Number', help='Unique number written on the vehicle motor (VIN/SN number)', copy=False)
     driver_id = fields.Many2one('res.partner', 'Driver', tracking=True, help='Driver of the vehicle', copy=False)
-    future_driver_id = fields.Many2one('res.partner', 'Future Driver', tracking=True, help='Next Driver of the vehicle', copy=False)
+    future_driver_id = fields.Many2one('res.partner', 'Future Driver', tracking=True, help='Next Driver of the vehicle', copy=False, domain="['|', ('company_id', '=', False), ('company_id', '=', company_id)]")
```

Comment:
@RomainLibert 
driver_id is used as a related field like 'car_id.driver_id' and if we use the domain in field declaration then it will produce an error on related field use.
ref : https://github.com/odoo/odoo/issues/34836

Thanks